### PR TITLE
Chore/revert limit and clazy fixes

### PR DESCRIPTION
**Pull request by caybro**
- revert `limit` property
- complete the previous clazy fixes (should result in speedups in some cases)

### CODE REVIEW

File: proxyroles/expressionrole.h [hunk 1]
```diff
@@ -23,15 +23,17 @@ class ExpressionRole : public SingleRole
 
 Q_SIGNALS:
     void expressionChanged();
-
+    
 private:
-    QVariant data(const QModelIndex& sourceIndex, const QQmlSortFilterProxyModel& proxyModel) override;
-    void updateContext(const QQmlSortFilterProxyModel& proxyModel);
+    QVariant extracted();
```
Comment (alexjba) [inline proxyroles/expressionrole.h]: where is this coming from? Can't find the implementation

Comment (caybro) [reply proxyroles/expressionrole.h]: Ah, overzealous "fix"; will revert that hunk

**Review by alexjba — APPROVED** [review]
LGTM!

File: proxyroles/expressionrole.h (hunk 1 above)
Comment (caybro) [reply proxyroles/expressionrole.h]: Reverted, thx for pointing out

**Pull request merged**